client/my-sites/site-settings/wrap-settings-form.jsx

```diff
@@ -0,0 +1,169 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { bindActionCreators } from 'redux';
+import { flowRight, omit, memoize } from 'lodash';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { protectForm } from 'lib/protect-form';
+import trackForm from 'lib/track-form';
+import {
+	isRequestingSiteSettings,
+	isSavingSiteSettings,
+	isSiteSettingsSaveSuccessful,
+	getSiteSettings
+} from 'state/site-settings/selectors';
+import { recordGoogleEvent, recordTracksEvent } from 'state/analytics/actions';
+import { saveSiteSettings } from 'state/site-settings/actions';
+import { removeNotice, successNotice, errorNotice } from 'state/notices/actions';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import QuerySiteSettings from 'components/data/query-site-settings';
+
+const wrapSettingsForm = getFormSettings => SettingsForm => {
+	class WrappedSettingsForm extends Component {
+		componentWillMount() {
+			this.props.replaceFields( getFormSettings( this.props.settings ) );
+		}
+
+		componentWillReceiveProps( nextProps ) {
+			if ( nextProps.siteId !== this.props.siteId ) {
+				nextProps.clearDirtyFields();
+			}
+
+			if ( nextProps.settings !== this.props.settings ) {
+				let newState = getFormSettings( nextProps.settings );
+				//If we have any fields that the user has updated,
+				//do not wipe out those fields from the poll update.
+				newState = omit( newState, nextProps.dirtyFields );
+				nextProps.replaceFields( newState );
+			}
+
+			if (
+			this.props.isSavingSettings &&
+			! nextProps.isSavingSettings
+			) {
+				if ( nextProps.isSaveRequestSuccessful ) {
+					nextProps.successNotice( nextProps.translate( 'Settings saved!' ), { id: 'site-settings-save' } );
+					nextProps.clearDirtyFields();
+					nextProps.markSaved();
+				} else {
+					let text;
+					switch ( nextProps.saveRequestError.error ) {
+						case 'invalid_ip':
+							text = nextProps.translate( 'One of your IP Addresses was invalid. Please try again.' );
+							break;
+						default:
+							text = nextProps.translate( 'There was a problem saving your changes. Please try again.' );
+					}
+					nextProps.errorNotice( text, { id: 'site-settings-save' } );
+				}
+			}
+		}
+
+		render() {
+			const utils = {
+				handleRadio: this.handleRadio,
+				handleSubmitForm: this.handleSubmitForm,
+				handleToggle: this.handleToggle,
+				onChangeField: this.onChangeField,
+				submitForm: this.submitForm,
+			};
+
+			return (
+				<div>
+					<QuerySiteSettings siteId={ this.props.siteId } />
+					<SettingsForm { ...this.props } { ...utils } />
+				</div>
+			);
+		}
+
+		// Some Utils
```

Comment:
Well! I like consistency between components, so I'm moving those above. My idea was to group these together but it can be done before render.